### PR TITLE
Clone ES query when stripping secrets

### LINKED ISSUE #893: sanitizer breaks my elasticsearch query
I have an elasticsearch query like this
```
{
  "aggs": {
    "type_count": {
      "cardinality": {
        "field": "type"
      }
    }
  }
}
```

before i install this agent, it works well.

and now the es server return a 400 bad request with this message

> Elasticsearch::Transport::Transport::Errors::BadRequest: [400] {"error":{"root_cause":[{"type":"parsing_exception","reason":"Expected [START_OBJECT] under [cardinality], but got a [VALUE_STRING] 

when i configure the option 'capture_elasticsearch_queries' to false,  the query gets fine

### PR DESCRIPTION
**Pull request by mikker**
Fixes #893

### CODE REVIEW

**Comment by Leihb** [comment]
be cautious . "dup" is shadow copy

**Comment by mikker** [comment]
@Leihb Good note! So something like ActiveSupport's `deep_dup` is necessary: https://github.com/rails/rails/blob/master/activesupport/lib/active_support/core_ext/object/deep_dup.rb

File: spec/elastic_apm/transport/filters/hash_sanitizer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+module ElasticAPM
+  module Transport
+    module Filters
+      RSpec.describe HashSanitizer do
+        let(:config) { Config.new }
+        subject { described_class.new(key_patterns: config.custom_key_filters + config.sanitize_field_names) }
```
Comment (estolfo) [inline spec/elastic_apm/transport/filters/hash_sanitizer_spec.rb:27]: might as well split this into two lines to save us one rubocop complaint :)

File: spec/elastic_apm/transport/filters/hash_sanitizer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            payload = { nested: { ApiKey: '123' } }
+
+            subject.strip_from!(payload)
+
+            expect(payload.dig(:nested, :ApiKey)).to eq '[FILTERED]'
+          end
+        end
+
+        describe '#strip_from' do
+          it 'returns a recursively clone copy' do
```
Comment (estolfo) [inline spec/elastic_apm/transport/filters/hash_sanitizer_spec.rb:58]:
```suggestion
          it 'returns a recursively cloned copy' do
```

File: spec/elastic_apm/util/deep_dup_spec.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+module ElasticAPM
+  RSpec.describe Util::DeepDup do
+    context 'hashes' do
+      it 'makes a copy' do
+        obj1 = Object.new
+        obj2 = Object.new
+        orig = { a: obj1, nested: { b: obj2 } }
+
+        expect(orig.dup[:a]).to be obj1
+        expect(orig.dup[:nested][:b]).to be obj2
```
Comment (estolfo) [inline spec/elastic_apm/util/deep_dup_spec.rb:32]: what's the point of testing these two lines?

Comment (mikker) [reply spec/elastic_apm/util/deep_dup_spec.rb:32]: I wanted to be sure but you're right. No point in testing Ruby itself.

File: spec/elastic_apm/util/deep_dup_spec.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
```
Comment (mikker) [inline spec/elastic_apm/util/deep_dup_spec.rb:33]:
```suggestion
  ```

File: spec/elastic_apm/transport/filters/hash_sanitizer_spec.rb (hunk 1 above)
Comment (mikker) [inline spec/elastic_apm/transport/filters/hash_sanitizer_spec.rb:27]:
```suggestion
        subject do
          described_class.new(
            key_patterns: config.custom_key_filters + config.sanitize_field_names
          )
        end
```

**Pull request merged**